hub-remote-deploy

```diff
@@ -0,0 +1,196 @@
+#!/bin/bash -e
+
+usage() {
+    cat << EOF
+
+Executes deployment of Certified Stack from kuberentes cluster
+
+Usage: 
+$ $(basename "$0") -c component1,component2
+Deploys deploys two components 
+
+Parameters:
+    -c --components       List of components to deploy
+    -o --offset           Deploy starting with the given component
+    -s  --silent          Suppress console outputs in favor of result codes
+    -p --no-precheck      Skip `hub ext aws status` command execution
+    -d --delete-toolbox   Deletes deployed toolbox 
+    -k --keep-toolbox     Keeps toolbox deployed on kubernetes cluster
+    --skip-guide          Suppress "What's next messages"
+    -V  --verbose         Verbose outputs for debug purpose
+    -h  --help            Print this message
+
+EOF
+}
+
+generate_manifests() {
+    trap "rm -rf $KUSTOMIZE_DIR" EXIT
+
+    envfile="$KUSTOMIZE_DIR/envfile"
+    env | grep -E '^(AWS_|GOOGLE_|AZURE_|TF_|TERM=|LANG=|LC_)' >"$envfile"
+    kustomize_manifest="$KUSTOMIZE_DIR/kustomization.yaml"
+
+    cat <<EOF > $kustomize_manifest
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  app: toolbox
+  provider: agilestacks.com
+configMapGenerator:
+- name: toolbox-env
+  namespace: $K8S_NAMESPACE
+  env: envfile
+- name: user-group
+  namespace: $K8S_NAMESPACE
+  literals:    
+  - USER=$USER
+  - UID=$(id -u)
+  - GID=$(id -g)
+resources:
+- deployment.yaml
+EOF
+
+pod_manifest="$KUSTOMIZE_DIR/deployment.yaml"
+cat <<EOF > $pod_manifest
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: toolbox
+  namespace: $K8S_NAMESPACE
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: toolbox-sa
+subjects:
+- kind: ServiceAccount
+  name: toolbox
+  namespace: $K8S_NAMESPACE
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Pod
```

Comment:
Any justification `Pod` vs `Job`? It might be that  `toolbox` `Pod` will stuck forever

Reply:
I considerate `job`, but then the job has to run in idle until `hub` has populated it with data, and then wait until hub download the data from job. In short: we need full control of life-cycle then we might as well use just a  pod. 
But I do agree that it would better for cleanup. 
I can update it to `job`.